src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -857,30 +918,42 @@ public CompilationInfo(Compilation compilation, bool hasSuccessfullyLoaded, Text
                                                     this.ProjectState.LanguageServices,
                                                     solution.Services,
                                                     cancellationToken));
+
+                                            // The count of trees was the same, but we must have had one generated tree disappear and another
+                                            // one appear at the same time. We can't reuse.
```

Comment:
I loved the "at least"/"equal number" but slightly redid the order -- since we're in the if block that's handling a _new_ file switched it to say:

    // The count of trees was the same, but something didn't match up. Since we're here, at least one tree
    // was added, and an equal number must have been removed. Rather than trying to incrementally update
    // this compilation, we'll just toss this and re-add all the trees.